projects/BFDR/BirthRecord_submissionProperties.cs

```diff
@@ -3521,5 +3519,275 @@ public string FatherEthnicityLiteral
+                }
+                AddReferenceToComposition(obs.Id, section);
+                Bundle.AddResourceEntry(obs, "urn:uuid:" + obs.Id);
+            }
+            // Create an empty quantity if needed
+            if (obs.Value == null || obs.Value as Quantity == null)
+            {
+                obs.Value = new Hl7.Fhir.Model.Quantity();
+            }
+            // Set the properties of the value individually to preserve any existing obs.Value.Extension entries
+            if (value != null)
+            {
+                (obs.Value as Quantity).Value = (int)value;
+                (obs.Value as Quantity).Unit = unit;
+                (obs.Value as Quantity).Code = unit;
+            }
+            return obs;
+        }
+
+        /// <summary>Mother's Prepregnancy Weight.</summary>
+        /// <value>the mother's prepregnancy weight in whole pounds, or -1 if explicitly unknown, or null if never specified</value>
+        /// <example>
+        /// <para>// Setter:</para>
+        /// <para>ExampleBirthRecord.MotherPrepregnancyWeight = 120;</para>
+        /// <para>// Getter:</para>
+        /// <para>Console.WriteLine($"Mother's Prepregancy Weight: {ExampleBirthRecord.MotherPrepregnancyWeight}");</para>
+        /// </example>
+        [Property("MotherPrepregnancyWeight", Property.Types.Int32, "Mother Prenatal", "Prepregnancy Weight.", false, BFDR.IGURL.ObservationMotherPrepregnancyWeight, true, 137)]
+        [FHIRPath("Bundle.entry.resource.where($this is Observation).where(code.coding.code='56077-1')", "")]
+        public int? MotherPrepregnancyWeight
+        {
+            // TODO replace codes with constants once BFDR value sets are autogenerated
+            get => GetWeight("56077-1");
+            set => SetWeight("56077-1", value, "lb_av", MOTHER_PRENATAL_SECTION, Mother.Id);
+        }
+
+        /// <summary>Mother's Weight at Delivery.</summary>
+        /// <value>the mother's weight at delivery in whole pounds, or -1 if explicitly unknown, or null if never specified</value>
+        /// <example>
+        /// <para>// Setter:</para>
+        /// <para>ExampleBirthRecord.MotherWeightAtDelivery = 120;</para>
+        /// <para>// Getter:</para>
+        /// <para>Console.WriteLine($"Mother's Weight at Delivery: {ExampleBirthRecord.MotherWeightAtDelivery}");</para>
+        /// </example>
+        [Property("MotherWeightAtDelivery", Property.Types.Int32, "Mother Prenatal", "Weight at Delivery.", false, BFDR.IGURL.ObservationMotherDeliveryWeight, true, 139)]
+        [FHIRPath("Bundle.entry.resource.where($this is Observation).where(code.coding.code='69461-2')", "")]
+        public int? MotherWeightAtDelivery
+        {
+            // TODO replace codes with constants once BFDR value sets are autogenerated
+            get => GetWeight("69461-2");
+            set => SetWeight("69461-2", value, "lb_av", MOTHER_PRENATAL_SECTION, Mother.Id);
```

Comment:
Should MotherWeightAtDelivery have category set to Vital Signs too? Looking at the Key Elements Table it looks like it should have it https://build.fhir.org/ig/HL7/fhir-bfdr/StructureDefinition-Observation-mother-delivery-weight.html

Reply:
Good catch, will fix.